python/test/unit/io/test_XDMF_P2.py

```diff
@@ -34,4 +34,4 @@ def test_read_write_p2_function(tempdir):
 
```

Comment:
Do we need another file? Can it all go in `test_XDMF.py`?

Reply:
I don't know why is it separated. Guess that @chrisrichardson wanted to have P2 test alone to run them faster(?). He'll probably remove/merge tests after higher order geometry is settled.

But anyway XDMF tests file is getting a bit bloated. From clarity reasons I'd suggest to split it.

Reply:
XDMF test file is bloated, but the fix is to clean it up rather than create a new file.